### ISSUE TITLE
Firebase Admin 12.1.1 issue with error.ts

### ISSUE THREAD

**Issue by Nushio** [firebase/firebase-admin-node#2571]
### [READ] Step 1: Are you in the right place?

  * For issues related to __the code in this repository__ file a Github issue.
  * If the issue pertains to Cloud Firestore, read the instructions in the "Firestore issue"
    template.
  * For general technical questions, post a question on [StackOverflow](http://stackoverflow.com/)
    with the firebase tag.
  * For general Firebase discussion, use the [firebase-talk](https://groups.google.com/forum/#!forum/firebase-talk)
    google group.
  * For help troubleshooting your application that does not fall under one
    of the above categories, reach out to the personalized
    [Firebase support channel](https://firebase.google.com/support/).

### [REQUIRED] Step 2: Describe your environment

  * Operating System version: MacOSX
  * Firebase SDK version: 12.2.1
  * Firebase Product: logger
  * Node.js version: 18.18.2
  * NPM version: 9.8.1
  
### [REQUIRED] Step 3: Describe the problem

Upgraded a (mature) codebase from 12.1.0 to 12.1.1, unit tests will no longer pass. 

> TypeError: Cannot read properties of undefined (reading 'message')

Narrowed down the issue to [src/utils/error.ts line 62](https://github.com/firebase/firebase-admin-node/blob/master/src/utils/error.ts)

If I add an optional to errorInfo?.message, and the tests pass now. 

This file hasn't changed in over 3 years according to the github history, so the fact that it's failing now might be related to something introduced in 12.1.1, probably https://github.com/firebase/firebase-admin-node/pull/2151

#### Steps to reproduce:

Upgrade to 12.1.1, and write a simple test using jest. 

Note I am using pnpm, but the error also occurs if you use npm: 

 at FirebaseError.get message [as message] (../node_modules/.pnpm/firebase-admin@12.1.1/node_modules/firebase-admin/lib/utils/error.js:46:31)
          at Array.forEach (<anonymous>)
          at Array.forEach (<anonymous>)
          at Array.forEach (<anonymous>)
      at Object.<anonymous> (../node_modules/.pnpm/firebase-functions@5.0.1_firebase-admin@12.1.1/node_modules/firebase-functions/lib/common/providers/firestore.js:25:19)
      at Object.<anonymous> (../node_modules/.pnpm/firebase-functions@5.0.1_firebase-admin@12.1.1/node_modules/firebase-functions/lib/v2/providers/firestore.js:32:21)
      at Object.<anonymous> (../node_modules/.pnpm/firebase-functions@5.0.1_firebase-admin@12.1.1/node_modules/firebase-functions/lib/v2/index.js:56:19)

#### Relevant Code:

  /** @returns The error message. */
  public get message(): string {
    return this.errorInfo.message;
  }

```
// TODO(you): code here to reproduce the problem
```

**Comment by aangelisc** [comment]
We are also experiencing this and there doesn't seem to be a good option for mocking this to avoid the problem.

**Comment by jbaldassari** [comment]
Same issue here trying to mock `firebase-admin` with `vitest`:

```
Error: [vitest] There was an error when mocking a module. If you are using "vi.mock" factory, make sure there are no top level variables inside, since this call is hoisted to top of the file. Read more: https://vitest.dev/api/vi.html#vi-mock
Caused by: TypeError: Cannot read properties of undefined (reading 'message')
```

Looking at the commits between 12.1.0 and 12.1.1, there were a couple of changes related to error classes, but nothing stands out to me as an obvious cause:
* https://github.com/firebase/firebase-admin-node/pull/2151
* https://github.com/firebase/firebase-admin-node/pull/2549

**Comment by Nushio** [comment]
Glad I'm not the only one. I didn't find an obvious culprit but making the message part optional does allow mocking to work, which seems to me like some firebase error classes are instanciated without a defined message error, like 

```
export class FirebaseDatabaseError extends FirebaseError {
constructor(info: ErrorInfo, message?: string) {
```

**Comment by Nushio** [comment]
It does for me. That's how I fixed it internally.

**Comment by lahirumaramba** [comment]
#2581 should fix it, but if you can provide us with a minimal repro that would be very useful for us to investigate and confirm the fix. Thanks!

**Comment by jbaldassari** [comment]
Thanks @lahirumaramba .  Here's a minimal reproduction case with 12.1.1: https://stackblitz.com/edit/firebase-admin-node-2571?file=test%2Fbasic.test.ts
![image](https://github.com/firebase/firebase-admin-node/assets/307312/832cd0ef-1b12-4c2e-bf1c-96a8e2bd88f2)

**Comment by Nushio** [comment]
I also tested the PR with our project and it works fine!

**Comment by lahirumaramba** [comment]
Thanks @jbaldassari for the repro and thanks @Nushio for confirming the fix. We will include this in the release next week.

**Comment by Nushio** [comment]
Hi @lahirumaramba , any eta for the release? We've been holding out from upgrading to 12.1.1 for about 2 weeks now.

**Comment by lahirumaramba** [comment]
We will include this in the release next week. Thanks for your patience!

**Comment by lahirumaramba** [comment]
The fix is now released in `v12.2.0`. Let us know if you run into any issues!

**Comment by Nushio** [comment]
Thank you! I'm deploying to staging, and will probably push this onto prod on Monday. We've been really looking forward the updated firestore library to see if it stabilizes our issues.
